core/node/libp2p/rcmgr.go

```diff
@@ -47,25 +45,12 @@ func ResourceManager(cfg config.SwarmConfig) func(fx.Lifecycle, repo.Repo) (netw
 
 			repoPath, err := config.PathRoot()
 			if err != nil {
-				return nil, opts, fmt.Errorf("error opening IPFS_PATH: %w", err)
+				return nil, opts, fmt.Errorf("opening IPFS_PATH: %w", err)
 			}
 
-			// Create limiter:
-			// - parse $IPFS_PATH/limits.json if exists
-			// - use defaultLimits from rcmgr_defaults.go
 			defaultLimits := adjustedDefaultLimits(cfg)
-			limitFilePath := filepath.Join(repoPath, NetLimitDefaultFilename)
-			limitFile, err := os.Open(limitFilePath)
-			switch {
-			case err == nil:
-				defer limitFile.Close()
-				limiter, err = rcmgr.NewLimiterFromJSON(limitFile, defaultLimits)
-				if err != nil {
-					return nil, opts, fmt.Errorf("error parsing libp2p limit file: %w", err)
-				}
-			case errors.Is(err, os.ErrNotExist):
-				limiter = rcmgr.NewStaticLimiter(defaultLimits)
-			default:
+			limiter, err := rcmgr.NewLimiter(*cfg.ResourceMgr.Limits, defaultLimits)
```

Comment:
This will panic if  `Swarm.ResourceMgr.Enabled` is `true` but  `Swarm.ResourceMgr.Limits` is missing from config JSON. 

Unfortunately, this is what all our users will do as the very first thing:
```console
$ ipfs config --json Swarm.ResourceMgr.Enabled true
$ ipfs daemon
Initializing daemon...
go-ipfs version: 0.13.0-dev-42b6071e4-dirty
Repo version: 12
System version: amd64/linux
Golang version: go1.18.1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x198516a]

goroutine 1 [running]:
github.com/ipfs/go-ipfs/core/node/libp2p.ResourceManager.func1({0x28c2f80, 0xc00011c418}, {0x3?, 0xd7e199?})
	src/github.com/ipfs/go-ipfs/core/node/libp2p/rcmgr.go:47 
```